### PR TITLE
Add option to invert the mouse movement when moving the map

### LINKED ISSUE #951: Add invert mouse option
Requested here: https://www.siedler25.org/index.php?com=forum&mod=forum&action=thread&id=1572&start=-1#last

There should be an option to invert the direction for map scrolling.

@Spikeone Which setting is "inverted"? Mouse moves right -> Map moves left (you see more on right) as it is currently and in S2 or the other way round?

Note: This function already exists, it is just missing an option entry

### PR DESCRIPTION
**Pull request by Flamefire**
Closes #951

@Spikeone Any idea for better naming or description of the option or is that fine?

- [x] Requires https://github.com/Return-To-The-Roots/s25client/pull/1554

### CODE REVIEW

**Comment by Spikeone** [comment]
> Closes #951
> 
> @Spikeone Any idea for better naming or description of the option or is that fine?

The current way is fine, although it might be user friendly to call it "original scrolling" which is off by default since most people seem to prefer another scroll direction. (So for them the controls are inverted already)

**Comment by Flamefire** [comment]
> "original scrolling" which is off by default since most people seem to prefer another scroll direction. (So for them the controls are inverted already)

Well I wanted to keep the spirit that we use the original behavior as the default. And the request was

>  Ohne invertiertes (rechte) Maustaste-Scrolling ...

So I guess "inverted" has already some notation which we can keep here :)

**Review by falbrechtskirchinger — COMMENTED** [review]
I don't like the wording. The tooltips are downright confusing.

File: libs/s25main/ingameWindows/iwSettings.cpp [hunk 1]
```diff
@@ -15,52 +15,64 @@
...
     }
+    AddText(ID_txtFullScreen, DrawPoint(15, 85), _("Mode:"), COLOR_YELLOW, FontStyle{}, NormalFont);
+    ctrlOptionGroup* optiongroup = AddOptionGroup(ID_grpFullscreen, GroupSelectType::Check);
+    optiongroup->AddTextButton(ID_btOn, DrawPoint(200, 70), Extent(150, 22), TextureColor::Grey, _("Fullscreen"),
+                               NormalFont);
+    optiongroup->AddTextButton(ID_btOff, DrawPoint(200, 95), Extent(150, 22), TextureColor::Grey, _("Windowed"),
+                               NormalFont);
+    optiongroup->SetSelection(SETTINGS.video.fullscreen); //-V807
+
+    AddCheckBox(ID_cbInvertMouse, DrawPoint(15, 124), Extent(150, 26), TextureColor::Grey, _("Invert mouse"),
```
Comment (falbrechtskirchinger) [inline libs/s25main/ingameWindows/iwSettings.cpp:63]: Would change it here as well. Once again title case, same as "Statistics Scale" below.

```suggestion
    AddCheckBox(ID_cbInvertMouse, DrawPoint(15, 124), Extent(150, 26), TextureColor::Grey, _("Invert Mouse Pan"),
```

File: libs/s25main/desktops/dskOptions.cpp [hunk 2]
```diff
@@ -238,6 +240,16 @@ dskOptions::dskOptions() : Desktop(LOADER.GetImageN("setup013", 0))
     }
     curPos.y += 50;
 
+    groupAllgemein->AddText(ID_txtInvertScroll, curPos, _("Mouse scroll behavior:"), COLOR_YELLOW, FontStyle{},
+                            NormalFont);
+    ctrlOptionGroup* invertScroll = groupAllgemein->AddOptionGroup(ID_grpInvertScroll, GroupSelectType::Check);
+    invertScroll->AddTextButton(ID_btOff, curPos + ctrlOffset, ctrlSize, TextureColor::Grey, _("Original"), NormalFont,
+                                _("When scrolling the map with the mouse move the camera."));
+    invertScroll->AddTextButton(ID_btOn, curPos + ctrlOffset2, ctrlSize, TextureColor::Grey, _("Inverted"), NormalFont,
+                                _("When scrolling the map with the mouse move the map."));
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply libs/s25main/desktops/dskOptions.cpp:249]: @Spikeone I'd consider you the expert on this ;-)   
From a quick google search both terms ("pan", "scroll") are used for map movement, so not sure which is correct.

However the change from Original/Inverted to On/Off is a good idea.

[the remark this answers is not in the document]
Comment (Spikeone) [reply libs/s25main/desktops/dskOptions.cpp:249]: @Flamefire I'd not consider myself as an expert, but "Scrolling" in games meant (at least in most old games) moving the view by moving the cursor to either screen side while panning meant using a mouse click to move the world around. So in RTTR/S2 we use panning I guess.

[the remark this answers is not in the document]
Comment (falbrechtskirchinger) [reply libs/s25main/desktops/dskOptions.cpp:249]: I had another look. Maybe "panning" is more at home in technical applications like 3D editors, etc. It's used in games too, but I guess "scrolling" is fine as well. The tooltip wording – especially "When scrolling the map with the mouse, move the map." – is still ambiguous.

So, the option can be named "Invert Mouse Scroll" or "Invert Mouse Pan".
My alternate suggestions for the tooltips would be:
"On": "Map moves in the opposite direction of mouse movement when scrolling/panning."
"Off": "Map moves in the same direction as mouse movement when scrolling/panning."

And if you want to keep the current tooltips, at least fix the missing commas before "move the map/camera."

Also, the button order should be flipped. "On" followed by "Off" seems more natural.

[the remark this answers is not in the document]
Comment (Flamefire) [reply libs/s25main/desktops/dskOptions.cpp]: > Also, the button order should be flipped. "On" followed by "Off" seems more natural.

I noticed we were inconsistent with that. We mostly do use On-Off but not on this screen. So fixed that too

Also used your suggestions for the tooltips with minor changes. See if that's good now.

**Review by falbrechtskirchinger — APPROVED** [review]
Looks good now.

Only one nitpick: when I wrote "scrolling/panning" I intended for you to pick one or the other matching the label. Both work too, I guess, but I'd have put "panning" first in that case. :man_shrugging:

**Comment by Flamefire** [comment]
@Flow86 ?

**Pull request merged**